// containers/uploadFiles/index.js
import React, {Component} from 'react';

import 'grommet-css';
import Form from 'components/uploadForm';

const URL = 'http://localhost:3777/';

class UploadForm extends Component {

    constructor(props) {
        super(props);

        this.state = {
            componentConfig: {
                iconFiletypes: ['.jpg', '.png', '.gif', '.folder'],
                showFiletypeIcon: true,
                postUrl: URL + 'upload'
            },

            djsConfig: {
                init: function () {
                    let output = {};
                    document.cookie.split(/\s*;\s*/).forEach((pair) => {
                        pair = pair.split(/\s*=\s*/);
                        output[pair[0]] = pair.splice(1).join('=');
                    });
                    this.on("sending", (file, xhr, data) => {
                        if (file.fullPath) {
                            data.append("path", file.fullPath);
                            data.append("dataCookies", JSON.stringify(output));
                        }
                    });
                }
            }
        }
    }

    render () {
        return (
            <div>
                <Form componentConfig = {this.state.componentConfig} djsConfig = {this.state.djsConfig} props = {this.props}/>
            </div>
        )
    };
}


export default UploadForm
